lib/pos_blockchain.go

```diff
@@ -1888,6 +1887,37 @@ func (bc *Blockchain) getUtxoViewAtBlockHash(blockHash BlockHash) (*UtxoView, er
 	return utxoView, nil
 }
 
+func (bc *Blockchain) AddSnapshotValidatorEntriesToCache(
+	snapshotAtEpochNumber uint64,
+	validatorEntries []*ValidatorEntry,
+) {
+	bc.cachedValidatorsBySnapshotAtEpoch.Add(snapshotAtEpochNumber, validatorEntries)
+}
+
+func (bc *Blockchain) GetAllSnapshotValidatorSetEntriesByStake(utxoView *UtxoView) ([]*ValidatorEntry, error) {
+	snapshotAtEpochNumber, err := utxoView.GetCurrentSnapshotEpochNumber()
+	if err != nil {
+		return nil, errors.Wrapf(err, "bc.GetAllSnapshotValidatorSetEntriesByStake: problem getting snapshotAtEpochNumber: ")
+	}
+	return bc.GetAllSnapshotValidatorSetEntriesByStakeAtEpochNumber(snapshotAtEpochNumber, utxoView)
+}
+
+func (bc *Blockchain) GetAllSnapshotValidatorSetEntriesByStakeAtEpochNumber(snapshotAtEpochNumber uint64, utxoView *UtxoView) (
```

Comment:
Rather than having this function taking any arbitrary UtxoView, it should construct its own UtxoView internally based on committed badger state only. That way, you don't run a risk of a newly created and uncommitted snapshot validator set getting added to the cache.
```suggestion
func (bc *Blockchain) GetAllSnapshotValidatorSetEntriesByStakeAtEpochNumber(snapshotAtEpochNumber uint64) (
```